src/SimpleExecutionController.cpp

```diff
@@ -239,27 +244,60 @@ void SimpleExecutionController::processEventCompoundJobCompletion(std::shared_pt
     std::string execution_host = (*(event->job->getActions().begin()))->getExecutionHistory().top().physical_execution_host;
 
     /* Remove all actions from memory and compute incremental output values in one loop */
-    double incr_compute_time = 0.;
+    double incr_compute_time = -9999.;
     double incr_infile_transfertime = 0.;
     double incr_infile_size = 0.;
     double incr_outfile_transfertime = 0.;
     double incr_outfile_size = 0.;
     double start_date = DBL_MAX;
-    double end_date = 0;
+    double end_date = DBL_MIN;
+    double hitrate = -9999.;
+
+    bool found_computation_action = false;
 
     // Figure out timings
     for (auto const &action : event->job->getActions()) {
-        double elapsed = action->getEndDate() - action->getStartDate();
-        WRENCH_DEBUG("Running action: %s, elapsed in s: %.2f", action->getName().c_str(), elapsed);
         start_date = std::min<double>(start_date, action->getStartDate());
         end_date = std::max<double>(end_date, action->getEndDate());
+        if (start_date < 0. || end_date < 0.) {
+            throw std::runtime_error(
+                "Start date " + std::to_string(start_date) +
+                " or end date " + std::to_string(end_date) + 
+                " of action " + action->getName() + " out of scope!"
+            );
+        }
+        double elapsed = end_date - start_date;
+        WRENCH_DEBUG("Analyzing action: %s, elapsed in s: %.2f", action->getName().c_str(), elapsed);
+
         // TODO: Better: Check for action type rather than doing string matching
```

Comment:
This TODO is accomplished by the line below, right? So this comment can be removed?

Reply:
Yes, I have removed it.

Reply:
thanks!